src/core/reactor.cc

```diff
@@ -1401,6 +1403,7 @@ void cpu_stall_detector::generate_trace() {
     auto delta = reactor::now() - _run_started_at;
 
     _total_reported++;
+    _stalls_histogram.add(delta);
```

Comment:
This is the wrong place to put the histogram increment.

 `generate_trace()` is rate limited. We have to ratelimit the logs out of necessity, but it would be undesirable to ratelimit histogram increments.

Also, `cpu_stall_detector` reports every long stall multiple times (as its millisecond duration crosses consecutive power of 2), which would render the histogram confusing at least.

I strongly suggest moving this line to the `if` inside `reactor::account_runtime`:
```
void
reactor::account_runtime(task_queue& tq, sched_clock::duration runtime) {
    if (runtime > (2 * _task_quota)) {
        tq._time_spent_on_task_quota_violations += runtime - _task_quota;
    }
    tq._vruntime += tq.to_vruntime(runtime);
    tq._runtime += runtime;
}
```

This way we both avoid the two problems I mentioned above (rate limiting and accounting the same stall multiple times), and we make the histogram count all stalls (as defined by `task_quota_ms`, not just the *logged* "stalls" as defined by `blocked-reactor-notify-ms`, which is much more useful.

Reply:
@michoecho thanks for the suggestion, I just forced push v3 with the changes